src/genie/libs/parser/iosxr/show_rib.py

```diff
@@ -0,0 +1,200 @@
+                }
+            }  
+
+class ShowRibTables(ShowRibTablesSchema):
+    """ Parser for show rib tables"""
+
+    
+    cli_command = ['show rib tables']
+
+    def cli(self,port="",output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+            
+        result_dict = {}
+        
+        p1 = re.compile(r'^(?P<vrf>\S+)\/(?P<table>\w+) +(?P<safi>\w+)'
+                        ' +(?P<table_id>\S+) +(?P<prfx_lmt>\d+) +(?P<prfx_cnt>\d+)'
+                        ' +(?P<tbl_ver>\d+) +(?P<pfx_notif>\w) +(?P<forw_refe>\w)'
+                        ' +(?P<tbl_del>\w) +(?P<tbl_conv>\w)')
+        
+        for line in out.splitlines():
+
+            if line:
+                line = line.strip()
+
+            else:
+                continue
+
+            m = p1.match(line)
+            
+            if m:
+                group = m.groupdict()
+                table_id=group['table_id']
+                result_dict[table_id]={}
+                result_dict[table_id].update({'vrf_name': group['vrf']})
+                result_dict[table_id].update({'table_name': group['table']})
+                result_dict[table_id].update({'safi': group['safi']})
+                result_dict[table_id].update({'table_id': group['table_id']})
+                result_dict[table_id].update({'prefix_limit': int(group['prfx_lmt'])})
+                result_dict[table_id].update({'prefix_count': int(group['prfx_cnt'])})
+                result_dict[table_id].update({'table_version': int(group['tbl_ver'])})
+                
+                if group['pfx_notif']=='N':
+                    result_dict[table_id].update({'prefix_limit_notified': 'No'})
+                elif group['pfx_notif']=='Y':
+                    result_dict[table_id].update({'prefix_limit_notified': 'Yes'})
+                    
+                if group['forw_refe']=='N':
+                    result_dict[table_id].update({'forward_referenced': 'No'})
+                elif group['forw_refe']=='Y':
+                    result_dict[table_id].update({'forward_referenced': 'Yes'})
+                    
+                if group['tbl_del']=='N':
+                    result_dict[table_id].update({'table_deleted': 'No'})
+                elif group['tbl_del']=='Y':
+                    result_dict[table_id].update({'table_deleted': 'Yes'})
+
+                if group['tbl_conv']=='N':
+                    result_dict[table_id].update({'table_reached_convergence': 'No'})
+                elif group['tbl_conv']=='Y':
+                    result_dict[table_id].update({'table_reached_convergence': 'Yes'})                                
+        
+        result_dict = {'table_id': result_dict}
```

Comment:
you can construct it from the beginning to avoid any confusion;
result_dict = {}
result_dict.setdefault('rib_tables', {}).setdefault('table_id', {})